### PR TITLE
Add a new hook for google datastore

### PR DESCRIPTION
**Pull request by mtagle**
Also included, adding a new connection type for google datastore,
and slightly correcting the gc base hook to reflect that SignedJwtAssertionCredentials initializer accepts a string or an iterable of strings for scope(s).

### CODE REVIEW

**Comment by mtagle** [comment]
proof for SignedJwtAssertionCredentials: https://google-api-python-client.googlecode.com/hg/docs/epy/oauth2client.client-pysrc.html#SignedJwtAssertionCredentials.__init__

File: airflow/contrib/hooks/datastore_hook.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    conn_name_attr = 'datastore_conn_id'
+
+    def __init__(self,
+                 scope=['https://www.googleapis.com/auth/datastore',
+                        'https://www.googleapis.com/auth/userinfo.email'],
+                 datastore_conn_id='google_cloud_datastore_default',
+                 delegate_to=None):
+        super(DatastoreHook, self).__init__(scope, datastore_conn_id, delegate_to)
+        # datasetId is the same as the project name
+        self.datasetId = self._extras_dejson().get('project')
```
Comment (criccomini) [inline airflow/contrib/hooks/datastore_hook.py]: prefer snake_case

**Comment by criccomini** [comment]
+1 Pending the one nit on snake_case

**Comment by criccomini** [comment]
+1

**Pull request merged**